frontend/javascripts/viewer/controller/segment_mesh_controller.ts

```diff
@@ -613,11 +640,51 @@ export default class SegmentMeshController {
 
       if (activeUnmappedSegmentId != null && containsSegmentId) {
         // Highlight (parts of) the mesh as active
-        this.updateMeshAppearance(obj, undefined, true, indexRange);
+        const highlightEntries =
+          indexRange !== null ? [{ range: indexRange, color: undefined }] : null;
+        this.updateMeshAppearance(obj, undefined, true, undefined, highlightEntries);
       } else if (obj.activeState) {
         // The mesh has an activeState, but that id is no longer
         // active. Therefore, clear it.
-        this.updateMeshAppearance(obj, undefined, false, null);
+        this.updateMeshAppearance(obj, undefined, false, undefined, null);
+      }
+    });
+  };
+
+  updateMinCutPartitionHighlighting = (minCutPartitions: MinCutPartitions | null) => {
+    this.meshesLayerLODRootGroup.traverse((_obj) => {
+      if (!("geometry" in _obj)) {
+        return;
+      }
+      // The cast is safe because MeshSceneNode adds only optional properties
+      const obj = _obj as MeshSceneNode;
+
+      const vertexSegmentMapping = obj.geometry.vertexSegmentMapping;
```

Comment:
I think this duplication with the function is okaisch but feel free to disagree :D